activation/atxdb.go

```diff
@@ -617,12 +633,20 @@ func (db *DB) GetNodeLastAtxID(nodeID types.NodeID) (types.ATXID, error) {
 	if exists := nodeAtxsIterator.Last(); !exists {
 		return *types.EmptyATXID, ErrAtxNotFound(fmt.Errorf("atx for node %v does not exist", nodeID.ShortString()))
 	}
-	return types.ATXID(types.BytesToHash(nodeAtxsIterator.Value())), nil
+
+	return types.ATXID(types.BytesToHash(nodeAtxsIterator.Value())), err
```

Comment:
This, and the defer func as written, won't work. This line should still `return ..., nil`. You need to turn the `error` return value into a named var (`err error`), then set it inside the defer func as you're doing on line 624. Here's more:

- https://yourbasic.org/golang/defer/#use-func-to-return-a-value
- See #3 in https://blog.learngoprogramming.com/5-gotchas-of-defer-in-go-golang-part-iii-36a1ab3d6ef1
- https://yourbasic.org/golang/named-return-values-parameters/

Reply:
you got this perfect [below](https://github.com/spacemeshos/go-spacemesh/pull/2176#discussion_r528021525) :)